app/scripts/controller-init/confirmations/ppom-controller-init.test.ts

```diff
@@ -0,0 +1,66 @@
+import { PPOMController } from '@metamask/ppom-validator';
+import { buildControllerInitRequestMock, CHAIN_ID_MOCK } from '../test/utils';
+import { PPOMControllerInit } from './ppom-controller-init';
+
+type PPOMControllerOptions = ConstructorParameters<typeof PPOMController>[0];
+
+jest.mock('@metamask/ppom-validator');
+
+function buildInitRequestMock() {
+  const requestMock = buildControllerInitRequestMock();
+
+  // @ts-expect-error Mocked subset of full state object
+  requestMock.getController.mockReturnValue({
+    state: { securityAlertsEnabled: true },
+  });
+
+  return requestMock;
+}
+
+describe('PPOM Controller Init', () => {
+  const ppomControllerClassMock = jest.mocked(PPOMController);
+
+  function testConstructorProperty<T extends keyof PPOMControllerOptions>(
+    property: T,
+    controllerProperties: Record<string, unknown> = {},
+  ): PPOMControllerOptions[T] {
+    const requestMock = buildInitRequestMock();
+
+    // @ts-expect-error Mocked subset of full state object
+    requestMock.getController.mockReturnValue({
```

Comment:
Nit: I found this confusing to read because it wasn't clear this was meant to return a mock preferences controller specifically. Maybe this would be clearer if we use a mock implementation here instead, and verify that the controller being asked for is `PreferencesController`?

Reply:
I see this same mock return value is in `buildInitRequestMock`. Maybe we can share the implementation

Reply:
Moved to a separate function and duplication removed.